### PR TITLE
boards: arm: Update efm32pg_stk3402a to run twister tests that pass

### PR DESCRIPTION
**Pull request by DRuffer-tmo**
Add information on twister and docs to index.rst.
Use only_tags: pearl_gecko to select passing tests. This allows selections to be very specific.
Give space to define various hardware maps.

### CODE REVIEW

**Comment by DRuffer-tmo** [comment]
This is PR 1 of many split off from [boards: arm: Update efm32pg_stk3402a and add tmo_dev_edge](https://github.com/zephyrproject-rtos/zephyr/pull/57424) to keep the discussions more focused.

File: boards/arm/efm32pg_stk3402a/doc/index.rst [hunk 1]
```diff
@@ -189,6 +189,77 @@ terminal session:
...
+convenience, until we push upstream. Then a more appropriate location
+can be found.
+
+I have a preference for
+`rich text editors <https://froala.com/blog/editor/a-beginners-guide-to-rich-text-editors/>`__\ ”
+and a promoter of
+`Literate Programming <https://en.wikipedia.org/wiki/Literate_programming>`__,
+but I’m using Microsoft’s Word to edit this text. I started from the
+original **rst** file and used `Pandoc <https://pandoc.org/>`__
+to convert to **docx**.
```
Comment (gmarull) [inline boards/arm/efm32pg_stk3402a/doc/index.rst:226]: what's going on? first, this doesn't seem to be related with the commit, and second, why do we need all this information in Zephyr docs?

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/arm/efm32pg_stk3402a/doc/index.rst (hunk 1 above)
Comment (DRuffer-tmo) [reply boards/arm/efm32pg_stk3402a/doc/index.rst:226]: You don't I just hadn't found a better location for it, but I can remove it from here, at least.

File: boards/arm/efm32pg_stk3402a/doc/index.rst [hunk 2]
```diff
@@ -189,6 +189,14 @@ terminal session:
...
    Hello World! arm
 
+Twister Testing on EFM32PG-STK3402A
+===================================
+
+The Twister tests can be run as follows:
+
+.. code-block:: console
+
+   scripts/twister --device-testing --hardware-map <relative path to>-map.yml
```
Comment (fkokosinski) [inline boards/arm/efm32pg_stk3402a/doc/index.rst:199]: IMO this is not needed in the board doc.

**Review by fkokosinski — CHANGES_REQUESTED** [review]
Hi, thanks again for splitting up your changes.

I don't think this is the right approach - you shouldn't need to specify your board manually for each testcase. Rather, you should filter out test cases based on preexisting tags. For example - you should filter out pm-related tests in the board yaml file if your board does not currently support power management, and not exclude your board manually in each of the testcases that use PM.

Also, `pearl_gecko` is not the right tag for these testcases - they are not pearl_gecko-specific.

In some cases you can add `exclude_platform` to a testcase, but this needs to be justified.

**Comment by DRuffer-tmo** [comment]
> Hi, thanks again for splitting up your changes.
> 
> I don't think this is the right approach - you shouldn't need to specify your board manually for each testcase. Rather, you should filter out test cases based on preexisting tags. For example - you should filter out pm-related tests in the board yaml file if your board does not currently support power management, and not exclude your board manually in each of the testcases that use PM.
> 
> Also, `pearl_gecko` is not the right tag for these testcases - they are not pearl_gecko-specific.
> 
> In some cases you can add `exclude_platform` to a testcase, but this needs to be justified.

I am not sure how familiar you are with the pearl gecko's history in Zephyr, but I do not believe twister tests were run on it when it was put into Zephyr in the 1st place. At this point, only a few random tests work and the failures have nothing to do with the preexisting tags. I started out with adding `exclude_platform` but that ended up being spam on almost every testsuite. I then used test plans, and you rightly pointed out that Zephyr does not use those in it's CI. So then I added this pearl_gecko tag to the 30 some that do pass, but they certainly do not fall into any preexisting groups, unless I simply eliminate all twister tests entirely.

If you are the antmicro person who is working with Todd.Dust@silabs.com then perhaps we need to put our heads together and figure out how to solve this problem together because Todd has already told me that he can only work on the twister tests that are preventing us from upstreaming our code and I have very few resources to do any of this for T-Mobile.

So where do we go from here?

File: boards/arm/efm32pg_stk3402a/doc/index.rst (hunk 2 above)
Comment (DRuffer-tmo) [reply boards/arm/efm32pg_stk3402a/doc/index.rst:199]: That depends on what we have to do to solve the problems that we have.
